### ISSUE TITLE
Take into account initial response type when deciding to send a continue in cursor constructor

### ISSUE THREAD

**Issue by deontologician** [rethinkdb/rethinkdb#5029]
Right now in Cursor.java, we call `maybeSendContinue` without looking at the initial response received. If the initial response is a `SUCCESS_SEQUENCE`, we should make sure `CONTINUE` is not sent.

See https://github.com/bchavez/RethinkDb.Driver/commit/274d80729625ee60f19b65d88e469f5f9a51aaea#commitcomment-14084487 for a discussion

**Comment by danielmewes** [comment]
We might have done this in the Python driver for better performance. The idea is that we send the continue before we start parsing the JSON response, in order to get better overall utilization of server and client resources.

**Comment by deontologician** [comment]
That's true for later batches, but the first batch is already parsed when this is called, otherwise we don't know to create a cursor at all

**Comment by danielmewes** [comment]
Ok saw your comment on that now:
> I think you're right. In the extend method, we call maybeContinue before we parse the response into json, since we want to parallelize parsing and the server getting the next batch ready. But in the constructor of the cursor, we've already had to parse the response to find out we need to create a cursor at all, so we always know the cursor response type before sending the continue query.

So I agree that we should change it to save the extra continue for such queries. Probably not a high priority though.

I'm also going to add the Python tag, since I suspect that we're doing the same thing there?

**Comment by adriantodt** [comment]
I implemented this without even knowing that this issue existed. This is now implemented to one of the branches (https://github.com/rethinkdb/rethinkdb-java/pull/12) of the Java driver, which now lives in a separated repository (https://github.com/rethinkdb/rethinkdb-java). This issue can be closed. Please, share feedback and discuss it on the PR :)
